src/acl.js

```diff
@@ -0,0 +1,84 @@
+export {Allowed} from './attribute/allowed';
+
+export class Acl {
+  permissions = {};
+
+  /**
+   * Set the permissions object with new permissions
+   *
+   * @param {object} permissions
+   *
+   * @chainable
+   */
+  setPermissions(permissions) {
+    this.permissions = {};
+
+    for (let resource in permissions) {
+      if (!permissions.hasOwnProperty(resource)) {
+        continue;
+      }
+
+      this.permit(resource, permissions[resource]);
+    }
+
+    return this;
+  }
+
+  /**
+   * @param {*} resource used to reference the permissions
+   * @param {*} rules used to set the value of the resource
+   *
+   * @chainable
+   */
+  permit(resource, rules) {
+    if (!this.permissions[resource]) {
+      this.permissions[resource] = {};
+    }
+
+    switch (true) {
+
+    case typeof rules === 'string':
+      this.permissions[resource][rules] = true;
+      break;
+
+    case ['boolean', 'object'].indexOf(typeof rules) > -1 && !Array.isArray(rules):
```

Comment:
i personally don't think that ` ['boolean', 'object'].indexOf(typeof rules)` is useful here. have it separated would increase readability imho